dali/test/python/operator/test_split_merge.py

```diff
@@ -118,7 +118,7 @@ def conditional_split_merge_reinterpret_pipe(dtype, layout, shape):
     input = fn.external_source(
         source=[[np.full((10, 10, 3), 42, dtype=np.int32) for _ in range(batch_size)]], cycle=True)
     pred = fn.external_source(
-        source=[[np.array(i % 2 == 0, dtype=np.bool) for i in range(batch_size)]], cycle=True)
+        source=[[np.array(i % 2 == 0, dtype=np.bool_) for i in range(batch_size)]], cycle=True)
```

Comment:
https://numpy.org/devdocs/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated
According to this, we can just use `bool` directly:
```suggestion
        source=[[np.array(i % 2 == 0, dtype=bool) for i in range(batch_size)]], cycle=True)
```

And it gives the same results:
```
>>> x = numpy.array(1, dtype=bool)
>>> y = numpy.array(1, dtype=numpy.bool_)
>>> x.dtype
dtype('bool')
>>> y.dtype
dtype('bool')
>>> x
array(True)
>>> y
array(True)
```

Reply:
Reworked